src/components/ExportDialog.tsx

```diff
@@ -71,9 +71,31 @@ const ExportModal = ({
       scale,
       shouldAddWatermark,
     });
-    previewNode?.appendChild(canvas);
+
+    let div: HTMLDivElement | undefined;
+    if (canvas) {
+      previewNode?.appendChild(canvas);
+    } else {
+      const h3 = document.createElement("h3");
+      const p = document.createElement("p");
+      div = document.createElement("div");
+      h3.innerText = "Canvas too big for PNG export";
+      // h3.innerText = t("alerts.CanvasTooBigForPNGExport");
+      p.innerHTML = `Cannot show preview image, nor export your drawing as PNG.
+      <br>
+      SVG Export might still work though.
+      <hr/>
+      <em>Tip</em>: try moving the farthest elements a bit closer together.`;
+      div.appendChild(h3);
+      div.appendChild(p);
+      previewNode?.appendChild(div);
```

Comment:
You may want to render this as a JSX prop. Either way, let's add these to `en.json` and use these via `t()` i18n helper.

Reply:
Do you mean using a prop for the message `<div>`? The problem is that only when `useEffect` kicks in we know that the canvas is too big. I also though about rendering the message in the canvas, but it seemed more complex than this solution.

As for the i18n helper, I have never used Crowdin so I wasn't sure where to put the text string and its translation. I'll fix it later.

Reply:
I think it makes sense to disable the `PNG` button if the canvas can't be exported as `.png`, but since `ToolButton` doesn't accept a `disabled` prop I didn't want to introduce new code.

Reply:
> Do you mean using a prop for the message <div>? 

Sorry, I wanted to say use something like this:

```
const [previewErrorMessage, setPreviewErrorMessage] = useState<null | JSX.Element>(null);

useEffect(() => {
	/* create canvas*/
	if (canvas) {
	   setPreviewErrorMessage(null);
	   /* render canvas imperatively as we do now */
	} else {
	  setPreviewErrorMessage(<>/* your error message stuff here */</>);
	}
}, [/*...*/]);

/*...*/
  return (
    <div className="ExportDialog">
      <div className="ExportDialog__preview" ref={previewRef}>
        {previewErrorMessage}
      </div>
```

The above could potentially have a render race condition, though (I'm not sure what react does to Node's children when you try to render `null` as the child -- but doesn't react just append instead of rewriting the node's existing content anyway? Mhm...).

> As for the i18n helper, I have never used Crowdin so I wasn't sure where to put the text string and its translation. I'll fix it later.

Just split the messages into separate strings and put them to `en.json` file, you don't need to set up anything else.

> I think it makes sense to disable the PNG button if the canvas can't be exported as .png, 

Agreed, that tripped me up as well.